dashboard/src/routes/TreeDetails/TreeDetailsFilter.tsx

```diff
@@ -1,104 +1,152 @@
 import FilterSummarySection from '@/components/Filter/SummarySection';
 import FilterCheckboxSection, {
   ICheckboxSection,
 } from '@/components/Filter/CheckboxSection';
-import {
-  TreeDetails as TreeDetailsType,
-  TTreeDetailsFilter,
-} from '@/types/tree/TreeDetails';
+import { TreeDetails as TreeDetailsType } from '@/types/tree/TreeDetails';
+
+type TFilterValues = { [key: string]: boolean };
+export type TFilter =
+  | { [key in TFilterKeys]: TFilterValues }
+  | Record<string, never>;
 
 interface ITreeDetailsFilter {
-  data?: TreeDetailsType;
-  onFilter: (filter: TTreeDetailsFilter) => void;
+  filter: TFilter;
+  onFilter: (filter: TFilter) => void;
 }
 
-type TFilterApplied = { [key: string]: boolean };
+export type TFilterKeys = (typeof filterFieldMap)[keyof typeof filterFieldMap];
+
+const filterFieldMap = {
+  git_repository_branch: 'branches',
+  config_name: 'configs',
+  architecture: 'archs',
+  valid: 'status',
+} as const;
+
+export const mapFilterToReq = (
+  filter: TFilter,
+): { [key: string]: string[] } => {
+  const filterMapped: { [key: string]: string[] } = {};
+
+  Object.entries(filterFieldMap).forEach(([reqField, field]) => {
+    const values = filter[field];
+    if (!values) return;
+
+    Object.entries(values).forEach(([value, isSelected]) => {
+      if (isSelected) {
+        if (!filterMapped[reqField]) filterMapped[reqField] = [];
+        filterMapped[reqField].push(value);
+      }
+    });
+  });
 
-const sanitizeData = (
-  data: TreeDetailsType | undefined,
-): [TFilterApplied, TFilterApplied, TFilterApplied, TFilterApplied, string] => {
-  const status = { TRUE: false, FALSE: false };
-  const branches: TFilterApplied = {};
-  const configs: TFilterApplied = {};
-  const archs: TFilterApplied = {};
-  let treeUrl = '';
+  return filterMapped;
+};
+
+export const createFilter = (data: TreeDetailsType | undefined): TFilter => {
+  const status = { True: false, False: false };
+  const branches: TFilterValues = {};
+  const configs: TFilterValues = {};
+  const archs: TFilterValues = {};
 
   if (data)
     data.builds.forEach(b => {
       if (b.git_repository_branch) branches[b.git_repository_branch] = false;
       if (b.config_name) configs[b.config_name] = false;
       if (b.architecture) archs[b.architecture] = false;
-      if (!treeUrl && b.git_repository_url) treeUrl = b.git_repository_url;
     });
 
-  return [status, branches, configs, archs, treeUrl];
+  return { status, branches, configs, archs };
 };
 
-const getFilterListFromObj = (filterObj: TFilterApplied): string[] =>
-  Object.keys(filterObj).filter(key => filterObj[key]);
+const changeFilterValue = (
+  filter: TFilter,
+  filterField: TFilterKeys,
+  value: string,
+  isSelected: boolean,
+): TFilter => {
+  const newFilter = { ...filter };
+  if (newFilter[filterField]) {
+    newFilter[filterField][value] = isSelected;
+  } else {
+    newFilter[filterField] = {
+      [value]: isSelected,
+    };
+  }
```

Comment:
your call, but what about using ternary operator?

Reply:
I prefer to use ternary only in expressions